func/internal/podevaluator.go

```diff
@@ -513,112 +526,175 @@ func (pm *podManager) retrieveOrCreatePod(ctx context.Context, image string, ttl
+							EmptyDir: &corev1.EmptyDirVolumeSource{},
+						},
 					},
 				},
 			},
-		},
+		}
+
+		return inlineBasePodTemplate, nil
 	}
-	// Server-side apply doesn't support name generation. We have to use Create
-	// if we need to use name generation.
-	if useGenerateName {
-		pod.GenerateName = podId + "-"
-		err = pm.kubeClient.Create(ctx, pod, client.FieldOwner(fieldManagerName))
-		if err != nil {
-			return client.ObjectKey{}, fmt.Errorf("unable to apply the pod: %w", err)
+}
+
+// Patches the expected port, and the original entrypoint of the kpt function into the function container
+func (pm *podManager) patchNewPodArgs(pod *corev1.Pod, de digestAndEntrypoint) error {
+	var patchedContainer bool
+	for _, container := range pod.Spec.Containers {
+		if container.Name == functionContainerName {
+			container.Args = append(container.Args,
+				"--port", defaultWrapperServerPort,
+				"--",
+			)
+			container.Args = append(container.Args, de.entrypoint...)
+			patchedContainer = true
 		}
-	} else {
-		pod.Name = podId
-		err = pm.kubeClient.Patch(ctx, pod, client.Apply, client.FieldOwner(fieldManagerName))
-		if err != nil {
-			return client.ObjectKey{}, fmt.Errorf("unable to apply the pod: %w", err)
+	}
+	if !patchedContainer {
+		return fmt.Errorf("haven't found the function container in the pod")
+	}
+	return nil
+}
+
+func (pm *podManager) patchNewPodImage(pod *corev1.Pod, image string) error {
+	var patchedContainer bool
+	for _, container := range pod.Spec.Containers {
+		if container.Name == functionContainerName {
+			container.Image = image
 		}
 	}
+	if !patchedContainer {
+		return fmt.Errorf("haven't found the function container in the pod")
+	}
+	return nil
+}
 
-	klog.Infof("created KRM function evaluator pod %v/%v for %q", pod.Namespace, pod.Name, image)
-	return client.ObjectKeyFromObject(pod), nil
+// Patch labels and annotations so the cache manager can keep track of the pod
+func (pm *podManager) patchNewPodMetadata(pod *corev1.Pod, ttl time.Duration, podId string) {
+	pod.ObjectMeta.Namespace = pm.namespace
+	annotations := pod.ObjectMeta.Annotations
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+	annotations[reclaimAfterAnnotation] = fmt.Sprintf("%v", time.Now().Add(ttl).Unix())
+	pod.ObjectMeta.Annotations = annotations
+
+	labels := pod.ObjectMeta.Labels
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	labels[krmFunctionLabel] = podId
+	pod.ObjectMeta.Labels = labels
 }
 
 // podIpIfRunningAndReady waits for the pod to be running and ready and returns the pod IP and a potential error.
 func (pm *podManager) podIpIfRunningAndReady(ctx context.Context, podKey client.ObjectKey) (ip string, e error) {
 	var pod corev1.Pod
 	// Wait until the pod is Running
-	if e := wait.PollImmediate(100*time.Millisecond, 60*time.Second, func() (done bool, err error) {
+	if e := wait.PollImmediate(100*time.Millisecond, pm.podReadyTimeout, func() (done bool, err error) {
```

Comment:
Not your doing but could we replace the deprecated wait function here?

`wait.PollImmediate is deprecated: This method does not return errors from context, use PollUntilContextTimeout. Note that the new method will no longer return ErrWaitTimeout and instead return errors defined by the context package. Will be removed in a future release.deprecated[default](https://pkg.go.dev/golang.org/x/tools/gopls/internal/analysis/deprecated)`